src/webgpu/api/operation/command_buffer/render/state_tracking.spec.ts

```diff
@@ -372,3 +368,269 @@ g.test('change_pipeline_before_and_after_vertex_buffer')
       );
     }
   });
+
+g.test('set_vertex_buffer_but_not_used_in_draw')
+  .desc(
+    `
+  Test that drawing after having set vertex buffer slots not used by the pipeline works correctly.
+  - In the test there are 2 draw calls in the render pass. The first draw call uses 2 vertex buffers
+    (position and color), and the second draw call only uses 1 vertex buffer (for color, the vertex
+    position is defined as constant values in the vertex shader). The test verifies if both of these
+    two draw calls work correctly.
+  `
+  )
+  .fn(async t => {
+    const kPositions = [-0.75, -0.25];
+    const kColors = [new Uint8Array([255, 0, 0, 255]), new Uint8Array([0, 255, 0, 255])];
+
+    // Initialize the vertex buffers with required vertex attributes (position: f32, color: f32x4)
+    const kAttribudeStride = 4;
+    const positionBuffer = t.device.createBuffer({
+      usage: GPUBufferUsage.VERTEX,
+      size: kAttribudeStride * kPositions.length,
+      mappedAtCreation: true,
+    });
+    t.trackForCleanup(positionBuffer);
+    const positions = new Float32Array(positionBuffer.getMappedRange());
+    positions.set(kPositions);
+    positionBuffer.unmap();
+
+    const colorBuffer = t.device.createBuffer({
+      usage: GPUBufferUsage.VERTEX,
+      size: kAttribudeStride * kPositions.length,
+      mappedAtCreation: true,
+    });
+    t.trackForCleanup(colorBuffer);
+    const colors = new Uint8Array(colorBuffer.getMappedRange());
+    colors.set(kColors[0]);
+    colors.set(kColors[1], kAttribudeStride);
+    colorBuffer.unmap();
```

Comment:
This one uses multiple `.set` calls but I think it would be OK to merge the two kColors Uint8Arrays into one and use `t.makeBufferWithContents` here too. (optional)